clouddq/classes/bigquery_client.py

```diff
@@ -202,3 +202,28 @@ def check_query_dry_run(self, query_string: str) -> None:
         except Forbidden as e:
             logger.error("User has insufficient permissions.")
             raise e
+
+    def is_table_exists(self, table: str) -> bool:
+        try:
+            client = self.get_connection()
+            client.get_table(table)
+            return True
+        except NotFound:
+            return False
+
+    def execute_query(self, query_string: str, job_config) -> bigquery.job.QueryJob:
+        """
+        The method is used to execute the sql query
+        Parameters:
+        query_string (str) : sql query to be executed
+        Returns:
+            result of the sql execution is returned
+        """
+        try:
+            client = self.get_connection()
+            logger.debug("Query string is \n %s", query_string)
+            query_job = client.query(query_string, job_config=job_config).result()
+            return query_job
+
+        except Exception as e:
+            logger.error(f"Query Execution failed with error {e}\n", exc_info=True)
```

Comment:
I think we can just raise the exception here and let the caller handle the error, instead of hiding the error behind a log statement here.

Reply:
in other words the try except clause may not be needed, unless you are going to specifically do something with the caught error

Reply:
Did this only for logging the error. But you are right, this try except clause can be removed as we aren't doing anything specific with the error.